### PR TITLE
Disable autocast cache during aotdispatch

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #84035
* #84011

### CODE REVIEW

**Review by jansel — APPROVED** [review]

File: torch/fx/experimental/proxy_tensor.py [hunk 1]
```diff
@@ -590,7 +611,8 @@ def wrap_fake_symbolic(x, sym_shape):
         else:
             func = f
 
-        with decompose(decomposition_table), fake_tensor_mode, sym_mode, proxy_mode:  # type: ignore[attr-defined]
+        with decompose(decomposition_table), fake_tensor_mode, \
+             sym_mode, proxy_mode, disable_autocast_cache():  # type: ignore[attr-defined]
```
Comment (ezyang) [inline torch/fx/experimental/proxy_tensor.py:615]: needs a comment saying why

**Pull request closed without merging**